sway-core/src/semantic_analysis/ast_node/declaration/function/function_parameter.rs

```diff
@@ -67,17 +72,13 @@ impl TypedFunctionParameter {
                     is_constant: IsConstant::No,
                     span: parameter.name.span(),
                 },
-                is_mutable: if parameter.is_mutable {
-                    VariableMutability::Mutable
-                } else {
-                    VariableMutability::Immutable
-                },
+                is_mutable: mutability.clone(),
```

Comment:
`VariableMutability` is just an enum, should probably be `Copy` so we don't need to explicitly `.clone()` it, but this is fine for now.